src/NuGet.Core/NuGet.Protocol/Plugins/PluginFactory.cs

```diff
@@ -144,16 +144,29 @@ public void Dispose()
             ConnectionOptions options,
             CancellationToken sessionCancellationToken)
         {
+            var args = string.Join(" ", arguments);
+#if IS_DESKTOP
             var startInfo = new ProcessStartInfo(filePath)
             {
-                Arguments = string.Join(" ", arguments),
+                Arguments = args,
                 UseShellExecute = false,
                 RedirectStandardError = false,
                 RedirectStandardInput = true,
                 RedirectStandardOutput = true,
                 StandardOutputEncoding = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false)
             };
-
+#else
+            var startInfo = new ProcessStartInfo
+            {
+                FileName = Environment.GetEnvironmentVariable("DOTNET_HOST_PATH"),
+                Arguments = $"\"{filePath}\" " + args,
+                UseShellExecute = false,
+                RedirectStandardError = false,
+                RedirectStandardInput = true,
+                RedirectStandardOutput = true,
```

Comment:
Where are you reading the standard output?

Reply:
The contract for the plugins is that they communicate with NuGet via STDIN/SDTOUT.